torch/nn/modules/linear.py

```diff
@@ -9,6 +10,15 @@
 from .lazy import LazyModuleMixin
 
 
+__all__ = [
```

Comment:
`NonDynamicallyQuantizableLinear` is not documented at https://pytorch.org/docs/master/nn.html#linear-layers
@jbschlosser is this meant to be a public API?

Reply:
I see `NonDynamicallyQuantizableLinear` listed in `test/allowlist_for_publicAPI.json` if that makes a difference. Happy to remove it from `__all__` if that's what we decide.

P.S. I'm surprised that adding type hints to these classes broke the newly added test, perhaps the test is a bit fragile? Or perhaps `__all__` should have been added to all files before the test went live?

Reply:
`NonDynamicallyQuantizableLinear` is not really intended to be public API, but the preceding underscore was mistakenly left out. Real fix would be to rename it, but that might be BC-breaking now. I say we should leave it out of `__all__` and keep it in the allowlist for now.

@anjali411 @albanD what's the reason adding type hints broke the test?

Reply:
I believe the error message was complaining about things like `typing.Any` not being part of PyTorch's public API. I would have expected `typing.*` to be ignored since it's a builtin library and just an import, not a class/function definition.

Reply:
The problem is that unless you specify the `__all__`, it is tricky for us to differentiate what comes from other libs or from here but isn't properly specified. So we are on the safe side here.